api_tests/collections/test_views.py

```diff
@@ -222,2261 +236,2635 @@ def test_create_bookmark_collection_with_wrong_title(self):
+        assert res.status_code == 403
+        assert 'detail' in res.json['errors'][0]
 
-    def test_requesting_node_returns_error(self):
-        node = NodeFactory(creator=self.user_one)
-        res = self.app.get(
+        #test_requesting_node_returns_error
+        node = NodeFactory(creator=user_one)
+        res = app.get(
             '/{}collections/{}/'.format(API_BASE, node._id),
-            auth=self.user_one.auth,
-            expect_errors=True
-        )
-        assert_equal(res.status_code, 404)
+            auth=user_one.auth,
+            expect_errors=True)
+        assert res.status_code == 404
 
+@pytest.mark.django_db
+class CollectionCRUDTestCase:
 
-class CollectionCRUDTestCase(ApiTestCase):
+    @pytest.fixture()
+    def user_two(self):
+        return AuthUserFactory()
 
-    def setUp(self):
-        super(CollectionCRUDTestCase, self).setUp()
-        self.user = AuthUserFactory()
-        self.user_two = AuthUserFactory()
+    @pytest.fixture()
+    def title_collection(self):
+        return 'Cool Collection'
 
-        self.title = 'Cool Collection'
-        self.new_title = 'Super Cool Collection'
+    @pytest.fixture()
+    def new_title_collection(self):
+        return 'Super Cool Collection'
 
-        self.collection = CollectionFactory(title=self.title, creator=self.user)
-        self.url = '/{}collections/{}/'.format(API_BASE, self.collection._id)
-        self.fake_url = '/{}collections/{}/'.format(API_BASE, '12345')
+    @pytest.fixture()
+    def collection(self, title_collection, user_one):
+        return CollectionFactory(title=title_collection, creator=user_one)
 
+    @pytest.fixture()
+    def url_collection_detail(self, collection):
+        return '/{}collections/{}/'.format(API_BASE, collection._id)
 
-def make_collection_payload(collection, attributes):
-    return {
-        'data': {
-            'id': collection._id,
-            'type': 'collections',
-            'attributes': attributes,
-        }
-    }
+    @pytest.fixture()
+    def url_fake_collection_detail(self, collection):
+        return '/{}collections/{}/'.format(API_BASE, '12345')
 
+    @pytest.fixture()
+    def payload_collection(self, collection):
+        def make_collection_payload(attributes):
+            return {
+                'data': {
+                    'id': collection._id,
+                    'type': 'collections',
+                    'attributes': attributes,
+                }
+            }
+        return make_collection_payload
 
+@pytest.mark.django_db
 class TestCollectionUpdate(CollectionCRUDTestCase):
 
-    def test_node_update_invalid_data(self):
-        res = self.app.put_json_api(self.url, "Incorrect data", auth=self.user.auth, expect_errors=True)
-        assert_equal(res.status_code, 400)
-        assert_equal(res.json['errors'][0]['detail'], "Malformed request.")
+    @pytest.fixture()
+    def logs_collection(self, collection):
```

Comment:
Rhiessen added a context manager, assert_latest_log, that can be used in place of the `assert_logs` decorator. Can you use that instead?

```python
def test_something(...):
     with assert_last_log(NodeLog.EDITED_TITLE, collection):
           # ... some operation
```